blog/2025-02-25-saml-idp-initiated-flow.md

```diff
@@ -0,0 +1,93 @@
+---
+title: "Keycloak SAML Identity Provider Idp-initiated flow with Okta"
+slug: keycloak-saml-identity-provider-idp-initiated-flow-with-okta
+date: 2025-02-25
+author: Phase Two
+tags: [phase_two, open_source, okta, authentication, authorization, keycloak, sso, idp-initiated]
+---
+
+### Idp-initiated flow
+
+When implementing SAML for the establishment of an Identity Provider, two primary options are available: Service Provider (SP)-initiated and Identity Provider (IdP)-initiated flows. The SP-initiated flow is widely recognized by users due to its straightforward configuration, which necessitates merely the exchange of some metadata. In contrast, the IdP-initiated flow is less intuitive and involves an additional step that may not be readily apparent to many users. The purpose of this blog is to elucidate the steps necessary to successfully execute the IdP-initiated flow.
+
+A fundamental understanding of SAML 2.0 and Keycloak is required to effectively follow the provided instructions.
+ 
+### Components
+
+The components involved in this flow are as follows:
+
+**Identity Provider:**
+Any identity Provider(IDP) that supports SAML 2.0 may be selected. The process begins by accessing the Identity Provider dashboard, where the user is prompted to authenticate. Upon successful authentication, the user may then request a service.
+
+**Keycloak SAML 2.0 Identity Provider:**
+The Keycloak Identity Provider use for identity brokering, will process the SAML Response received from the Identity Provider. It is responsible for operations such as provisioning, signature verification, decryption etc.
+
+**Keycloak Realm Client:**
+The Keycloak SAML client function is to maintain the authenticated user session within Keycloak. Another function of this generic client is to forward the authenticated user to the Service Provider.
+
+**Service Provider:**
+The Service Provider refers to the application that the user seeks to access. Once the user has been authenticated in Keycloak, a new SAML Response is generated by the realm client and subsequently consumed by the Service Provider.
+
```

Comment:
In other articles, we also prompt the user to set up a keycloak instance, and give them instructions on how to set it up. E.g. https://github.com/p2-inc/phasetwo-docs/blob/main/blog/2024-10-10-oidc-spa.mdx?plain=1#L34-L39